eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureClientPlugin.cs

```diff
@@ -46,6 +47,19 @@ public override void Configure()
     {
         base.Configure();
         AddMetadataReference(MetadataReference.CreateFromFile(typeof(Response).Assembly.Location));
+        AddSharedSourceDirectory(Path.Combine(GetRepositoryRootPath(), "sdk/core/Azure.Core/src/Shared"));
+    }
+
+    private string GetRepositoryRootPath()
```

Comment:
If we copy the shared source to the output directory, e.g. like https://github.com/microsoft/typespec/blob/main/packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Microsoft.Generator.CSharp.Tests.csproj#L32-L36, then we wouldn't need to get the repo root nor would we need to rely on generating in the repo. We could instead use the Assembly.Location like we do for the Metadata references.

Reply:
I don't copy the shared code to the output library because:
- Azure SDKs do live within azure-sdk-for-net now
- It's a waste to copy them to output library since they can be found from the original place, like we do not copy them into Azure.Generator just link to the original files, there could be more share files

Reply:
@live1206:  We need to be able to support the scenario of generating or regenerating outside of the context of the Azure SDK repository.

Reply:
@jsquire I can copy them to output for this one. But if we want to be able to generate outside of azure-sdk-for-net we need to copy the shared code to Azure.Generator if it lies outside this repo.

Reply:
updated.